File: backend/src/main/java/edu/cit/daal/techtrack/dto/response/AssetResponse.java
package edu.cit.daal.techtrack.dto.response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class AssetResponse {

    private Long id;
    private String name;
    private String category;
    private String description;
    private String serialNumber;
    private String assetTag;
    private String status;
    private List<ImageDto> images;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Data
    @Builder
    public static class ImageDto {
        private Long id;
        private String filePath;
        private boolean primary;
    }
}

File: Mobile/app/src/main/java/com/techtrack/inventory/data/remote/model/response/AssetResponse.kt
package com.techtrack.inventory.data.remote.model.response

import com.google.gson.annotations.SerializedName

data class AssetResponse(
    val id: Long,
    val name: String,
    val category: String,
    val description: String?,
    val serialNumber: String?,
    val assetTag: String,
    val status: String,
    val images: List<AssetImageDto>,
    val createdAt: String?,
    val updatedAt: String?
) {
    fun primaryImagePath(): String? = images.firstOrNull { it.primary }?.filePath
        ?: images.firstOrNull()?.filePath
}

data class AssetImageDto(
    val id: Long,
    val filePath: String,
    @SerializedName("primary") val primary: Boolean
)
